macros/sql/fetch.sql

```diff
@@ -0,0 +1,21 @@
+{% macro fetch(sql) %}
+{# This macro returns a dictionary of the form {column_name: [results]} #}
+
+    {%- call statement('get_sql_results', fetch_result=True) -%}
+
+        {{sql}}
+
+    {%- endcall -%}
+    
+    {% set sql_results={} %}
+    
+    {%- if execute -%}
+        {% set sql_results_table = load_result('get_sql_results').table.columns %}
+        {% for column, column_values in sql_results_table.items() %}
```

Comment:
I was rushing out the door last time I was looking at this. I think this is a slightly nicer naming (tbh I probably named these in the first place)
```suggestion
        {% for column_name, column in sql_results_table.items() %}
        {% do sql_results.update({column_name: columns.values()}) %}
```